### PR TITLE
Rocm support

### PR DESCRIPTION
**Pull request by ItsPitt**

### CODE REVIEW

**Comment by ItsPitt** [comment]
Not sure why it's showing all of those commits. It was only for the last one.

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 1]
```diff
@@ -87,7 +87,7 @@ struct LoadCode32<3> {
             uint8_t* p,
             int offset) {
         p += offset * 3;
-	using T = uint8_t __attribute__((ext_vector_type(3)));
+    using T = uint8_t __attribute__((ext_vector_type(3)));
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:90]: can you revert the white space change?

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 2]
```diff
@@ -129,7 +129,7 @@ struct LoadCode32<12> {
             uint8_t* p,
             int offset) {
         p += offset * 12;
-	using T = uint32_t __attribute__((ext_vector_type(3)));
+    using T = uint32_t __attribute__((ext_vector_type(3)));
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:132]: same here

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 3]
```diff
@@ -143,7 +143,7 @@ struct LoadCode32<16> {
             uint8_t* p,
             int offset) {
         p += offset * 16;
-	using T = uint32_t __attribute__((ext_vector_type(4)));
+    using T = uint32_t __attribute__((ext_vector_type(4)));
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:146]: and here

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 4]
```diff
@@ -157,7 +157,7 @@ struct LoadCode32<20> {
             uint8_t* p,
             int offset) {
         p += offset * 20;
-	using T = uint32_t __attribute__((ext_vector_type(5)));
+    using T = uint32_t __attribute__((ext_vector_type(5)));
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:160]: and here

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 5]
```diff
@@ -171,10 +171,10 @@ struct LoadCode32<24> {
             uint8_t* p,
             int offset) {
         p += offset * 24;
-	using T = uint32_t __attribute__((ext_vector_type(6)));
-	T* t = reinterpret_cast<T*>(p);
+    using T = uint32_t __attribute__((ext_vector_type(6)));
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:174]: and here

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 6]
```diff
@@ -171,10 +171,10 @@ struct LoadCode32<24> {
...
+    T* t = reinterpret_cast<T*>(p);
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:175]: here

File: faiss/gpu/impl/PQCodeLoad.cuh [hunk 7]
```diff
@@ -171,10 +171,10 @@ struct LoadCode32<24> {
...
         T* u = reinterpret_cast<T*>(code32);
-	u[0] = __builtin_nontemporal_load(t);
+    u[0] = __builtin_nontemporal_load(t);
```
Comment (iotamudelta) [inline faiss/gpu/impl/PQCodeLoad.cuh:177]: here

File: faiss/gpu/utils/LoadStoreOperators.cuh [hunk 8]
```diff
@@ -55,7 +55,7 @@ template <>
 struct LoadStore<Half8> {
     static inline __device__ Half8 load(void* p) {
         Half8 out;
-	Half8* t = reinterpret_cast<Half8*>(p);
+        Half8* t = reinterpret_cast<Half8*>(p);
```
Comment (iotamudelta) [inline faiss/gpu/utils/LoadStoreOperators.cuh:58]: here

**Review by iotamudelta — CHANGES_REQUESTED** [review]
the actual change in IVFFlat LGTM. please revert the white space only changes in the other two files

**Comment by iotamudelta** [comment]
> Not sure why it's showing all of those commits. It was only for the last one.

and I'll just squash and merge - so the extra commits will go away

**Comment by ItsPitt** [comment]
Those white space changes were done because the code I pulled over from the other branch had tabs in those spots. While the rest of the file had spaces. I shared those to match the rest of the file.

**Pull request merged**